src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionNotificationsRetryQueueConsumerTest.kt

```diff
@@ -253,6 +248,154 @@ class TransactionNotificationsRetryQueueConsumerTest {
+        .willAnswer { Mono.just(it.arguments[0]) }
+      given(transactionRefundRepository.save(capture(transactionRefundEventStoreCaptor)))
+        .willAnswer { Mono.just(it.arguments[0]) }
+      given(paymentGatewayClient.requestVPosRefund(any()))
+        .willReturn(
+          Mono.just(VposDeleteResponseDto().status(VposDeleteResponseDto.StatusEnum.CANCELLED)))
+
+      given(transactionsViewRepository.findByTransactionId(TRANSACTION_ID))
+        .willReturnConsecutively(
+          listOf(
+            Mono.just(
+              transactionDocument(TransactionStatusDto.NOTIFICATION_ERROR, ZonedDateTime.now())),
+            Mono.just(transactionDocument(TransactionStatusDto.EXPIRED, ZonedDateTime.now())),
+            Mono.just(
+              transactionDocument(TransactionStatusDto.REFUND_REQUESTED, ZonedDateTime.now()))))
+      StepVerifier.create(
+          transactionNotificationsRetryQueueConsumer.messageReceiver(
+            BinaryData.fromObject(notificationErrorEvent).toBytes(), checkpointer))
+        .expectNext()
+        .verifyComplete()
+      verify(checkpointer, times(1)).success()
+      verify(transactionsEventStoreRepository, times(1))
+        .findByTransactionIdOrderByCreationDateAsc(TRANSACTION_ID)
+      verify(notificationsServiceClient, times(1)).sendNotificationEmail(any())
+      verify(notificationRetryService, times(0)).enqueueRetryEvent(any(), any())
+      verify(transactionsViewRepository, times(3)).save(any())
+      verify(transactionRefundRepository, times(2)).save(any())
+      verify(paymentGatewayClient, times(1)).requestVPosRefund(any())
+      verify(transactionUserReceiptRepository, times(1)).save(any())
+      verify(refundRetryService, times(0)).enqueueRetryEvent(any(), any())
+      verify(userReceiptMailBuilder, times(1)).buildNotificationEmailRequestDto(baseTransaction)
+      val savedEvent = transactionUserReceiptCaptor.value
+      assertEquals(TransactionEventCode.TRANSACTION_USER_RECEIPT_ADDED_EVENT, savedEvent.eventCode)
+      assertEquals(transactionUserReceiptData, savedEvent.data)
+      val expectedStatuses =
+        listOf(
+          TransactionStatusDto.NOTIFIED_KO,
+          TransactionStatusDto.REFUND_REQUESTED,
+          TransactionStatusDto.REFUNDED)
+      val expectedEventCodes =
+        listOf(
+          TransactionEventCode.TRANSACTION_REFUND_REQUESTED_EVENT,
+          TransactionEventCode.TRANSACTION_REFUNDED_EVENT)
+      expectedEventCodes.forEachIndexed { index, eventCode ->
+        assertEquals(eventCode, transactionRefundEventStoreCaptor.allValues[index].eventCode)
+        assertEquals(
+          TransactionStatusDto.NOTIFICATION_ERROR,
+          transactionRefundEventStoreCaptor.allValues[index].data.statusBeforeRefunded)
+      }
+      expectedStatuses.forEachIndexed { index, transactionStatus ->
+        assertEquals(transactionStatus, transactionViewRepositoryCaptor.allValues[index].status)
+      }
+    }
```

Comment:
Add test for legacy TransactionUserReceiptAddRetriedEvent event too

Reply:
Fixed in 21230aa